pycrostates/segmentation/transitions.py

```diff
@@ -0,0 +1,197 @@
+    """
+    # common error checking
+    _check_value(stat, ("probability", "proportion", "percent"), "stat")
+    _check_type(ignore_self, (bool,), "ignore_self")
+
+    # reshape if epochs (returns a view)
+    labels = labels.reshape(-1)
+    states = np.arange(-1, n_clusters)
+    # expected probabilities
+    T_expected = np.zeros(shape=(states.size, states.size))
+    for state_from in states:
+        n_from = np.sum(labels == state_from)  # no state_from in labels
+        for state_to in states:
+            n_to = np.sum(labels == state_to)
+            if n_from != 0:
+                if state_from != state_to:
+                    T_expected[state_from, state_to] = n_to
+                else:
+                    T_expected[state_from, state_to] = n_to - 1
+            else:
+                T_expected[state_from, state_to] = 0
+
+    # ignore unlabeled
+    T_expected = T_expected[:-1, :-1]
+    if ignore_self:
+        np.fill_diagonal(T_expected, 0)
+    # transform to probability
+    with np.errstate(divide="ignore", invalid="ignore"):
+        T_expected = T_expected / T_expected.sum(axis=1, keepdims=True)
+        np.nan_to_num(T_expected, nan=0, posinf=0, copy=False)
+    if stat == "percent":
+        T_expected = T_expected * 100
+
+    return T_expected
+
+
+def _check_labels_n_clusters(
+    labels: NDArray[int],
+    n_clusters: int,
+) -> None:
+    """Checker for labels and n_clusters."""
+    _check_type(labels, (np.ndarray,), "labels")
+    _check_type(n_clusters, ("int",), "n_clusters")
+    if n_clusters <= 0:
+        raise ValueError(
+            "The provided number of clusters 'n_clusters' must be a strictly "
+            f"positive integer. '{n_clusters}' is invalid."
+        )
+
+    # only accept array of integers to simplify maintenance
+    if labels.dtype != "int":
+        raise ValueError(
+            "The argument 'labels' must contain the labels of each timepoint "
+            "encoded as consecutive positive integers (0-indexed). Make sure "
+            f"you are providing an integer array. '{labels.dtype}' is "
+            "invalid."
+        )
+    # check for negative integers except -1
+    if np.any(labels < -1):
+        raise ValueError(
+            "The argument 'labels' must contain the labels of each timepoint "
+            "encoded as consecutive positive integers (0-indexed) with '-1' "
+            "representing unlabelled segments. Negative integers except -1 "
+            "are invalid."
+        )
+    # check that the labels are in the range of clusters
+    states = sorted([elt for elt in np.unique(labels) if 0 <= elt])
+    if not np.all(np.isin(states, np.arange(n_clusters))):
+        raise ValueError(
+            "The argument 'labels' must contain the labels of each timepoint "
+            "encoded as consecutive positive integers (0-indexed), between 0 "
+            f"and 'n_clusters' ({n_clusters}. '{states}' are invalid."
```

Comment:
This error message seems a bit misleading: ` '{states}' are invalid.` will print both allowed and non-allowed states. I would rather print only non-allowed states are rephrase/delete the last part of the error message

Reply:
Done.